sdkconfig.defaults

```diff
@@ -24,3 +25,33 @@ CONFIG_ESP_SECURE_CERT_SUPPORT_LEGACY_FORMATS=y
 CONFIG_GRI_SUB_PUB_UNSUB_DEMO_MAX_COMMAND_SEND_BLOCK_TIME_MS=10000
 CONFIG_GRI_SUB_PUB_UNSUB_DEMO_DELAY_BETWEEN_SUB_PUB_UNSUB_LOOPS_MS=10000
 CONFIG_GRI_MQTT_AGENT_CONNACK_RECV_TIMEOUT_MS=10000
+
+# Set optimization for size
+CONFIG_COMPILER_OPTIMIZATION_SIZE=y
+
+# Set default FreeRTOS Stack sizes
+CONFIG_FREERTOS_TIMER_TASK_STACK_DEPTH=1536
+CONFIG_FREERTOS_IDLE_TASK_STACKSIZE=1024
+
+# LWIP config
+CONFIG_LWIP_MAX_SOCKETS=8
+
+# AWS OTA
+CONFIG_LOG2_FILE_BLOCK_SIZE=12
+CONFIG_MAX_NUM_BLOCKS_REQUEST=8
+CONFIG_MAX_NUM_OTA_DATA_BUFFERS=8
+CONFIG_ALLOW_DOWNGRADE=0
+CONFIG_OTA_DATA_OVER_MQTT=y
+# CONFIG_OTA_DATA_OVER_HTTP is not set
+CONFIG_OTA_DATA_OVER_MQTT_PRIMARY=y
+CONFIG_OTA_PRIMARY_DATA_PROTOCOL=1
```

Comment:
Just out of concern, Where are you getting these numbers from?

Reply:
I got them by running `idf.py menuconfig` changing the settings a bit to shrink the image, and then copying them from the `sdkconfig` that gets generated